test/k4FWCoreTest/src/components/ExampleFunctionalTransformerMultiple.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2014-2023 Key4hep-Project.
+ *
+ * This file is part of Key4hep.
+ * See https://key4hep.github.io/key4hep-doc/ for further info.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Gaudi/Property.h"
+#include "GaudiAlg/Transformer.h"
+
+#include "edm4hep/MCParticleCollection.h"
+#include "edm4hep/SimTrackerHitCollection.h"
+#include "edm4hep/TrackCollection.h"
+#include "edm4hep/TrackerHitCollection.h"
+#include "podio/UserDataCollection.h"
+
+// Define BaseClass_t
+#include "k4FWCore/BaseClass.h"
+
+#include <string>
+
+// Which type of collection we are reading
+using FloatColl         = podio::UserDataCollection<float>;
+using ParticleColl      = edm4hep::MCParticleCollection;
+using SimTrackerHitColl = edm4hep::SimTrackerHitCollection;
+using TrackerHitColl    = edm4hep::TrackerHitCollection;
+using TrackColl         = edm4hep::TrackCollection;
+
+// As a simple example, we'll write an integer and a collection of MCParticles
+using Counter_t  = podio::UserDataCollection<int>;
+using Particle_t = edm4hep::MCParticleCollection;
```

Comment:
Same here as above, unless making the in- and outputs follow different conventions was a conscious choice?